src/main/java/com/google/devtools/build/lib/packages/TargetUtils.java

```diff
@@ -239,6 +253,47 @@ private static boolean hasConstraint(Rule rule, String keyword) {
     return ImmutableMap.copyOf(map);
   }
 
+  /**
+   * Returns the execution info, obtained from the rule's tags and the execution requirements provided.
+   * Only supported tags are included into the execution info,
+   * see {@link #LEGAL_EXEC_INFO_KEYS} and {@link #TAGS_PROPAGATED_TO_EXEC_INFO}.
+   */
+  public static Map<String, String> getFilteredExecutionInfo(Object executionRequirementsUnchecked,
+      Rule rule) throws EvalException {
+    Map<String, String> executionInfo = Maps.newLinkedHashMap();
+    executionInfo.putAll(getExecutionInfoFromTags(rule));
+
+    if (executionRequirementsUnchecked != Runtime.NONE) {
```

Comment:
Ignoring my comment above to validate the input before entering this method, do we need to check for None at all when we call castSkylarkDictOrNoneToDict anyway?

Reply:
true, we don't need it  at all